src/buffer/out/textBuffer.cpp

```diff
@@ -2146,6 +2134,34 @@ std::string TextBuffer::GenRTF(const TextAndColor& rows, const int fontHeightPoi
     }
 }
 
+void TextBuffer::_AppendRTFText(std::ostringstream& contentBuilder, const std::wstring_view& text)
+{
+    for (const auto codeUnit : text)
+    {
+        if (codeUnit <= 127)
+        {
+            switch (codeUnit)
+            {
+            case L'\\':
+            case L'{':
+            case L'}':
+                contentBuilder << "\\" << gsl::narrow<char>(codeUnit);
+                break;
+            default:
+                contentBuilder << gsl::narrow<char>(codeUnit);
+            }
+        }
+        else if (codeUnit <= 32767)
+        {
+            contentBuilder << "\\u" << std::to_string(codeUnit) << "?";
+        }
+        else
+        {
+            contentBuilder << "\\u" << std::to_string(codeUnit - 65536) << "?";
+        }

```

Comment:
Since C++20, signed integers are defined to use two's complement just like unsigned integers (P0907R4).
You can rewrite this code as:
```cpp
else
{
    // Windows uses unsigned wchar_t - RTF uses signed ones.
    contentBuilder << "\\u" << std::to_string(til::bit_cast<int16_t>(codeUnit)) << "?";
}
```

Reply:
Done